master/pkg/schemas/merge.go

```diff
@@ -84,9 +85,15 @@ func merge(obj reflect.Value, src reflect.Value, name string) reflect.Value {
 		return merge(obj.Elem(), src.Elem(), name)
 	}
 
-	// Detect Mergables.
-	if mergeable, ok := obj.Interface().(Mergable); ok {
-		return reflect.ValueOf(mergeable.Merge(src.Interface()))
+	// Detect Mergables, but disallow pointers.  The reason is that if you have a method like:
+	//    func (t Thing) Merge(other interface{}) interface{}
+	// then Merge() will return a plain Thing object, but a pointer to a Thing will still be treated
+	// as mergable.  Then, schemas.Merge(&t, &t2).(*Thing) would panic because it returns the wrong
+	// type.
+	if obj.Kind() != reflect.Ptr {
```

Comment:
q: if we have a mergable and a non-pointer here, should we panic instead?

Reply:
`obj.Kind() != reflect.Ptr` == brad cant read

Reply:
obj != src == brad cant read

Reply:
We have other checks to make sure the types are exactly right.  Also, we don't modify either of `src` or `obj` when we merge, we return a clean copy.